tests/jwt_validate.c

```diff
@@ -0,0 +1,440 @@
+
+	ck_assert_int_eq(0, jwt_validate(jwt, jwt_valid));
+
+	// jwt is invalid when now >= expires
+	ret = jwt_valid_set_now(jwt_valid, not_before);
+	ck_assert_int_eq(ret, 0);
+
+	ck_assert_int_eq(1, jwt_validate(jwt, jwt_valid));
+
+	jwt_valid_free(jwt_valid);
+	__teardown_jwt();
+}
+END_TEST
+
+START_TEST(test_jwt_valid_expires)
+{
+	jwt_valid_t *jwt_valid = NULL;
+	int ret = 0;
+
+	__setup_jwt();
+	jwt_add_grant_int(jwt, "exp", expires);
+
+	ret = jwt_valid_new(&jwt_valid, JWT_ALG_NONE);
+	ck_assert_int_eq(ret, 0);
+	ck_assert(jwt_valid != NULL);
+
+	// jwt is valid when now < expires
+	ret = jwt_valid_set_now(jwt_valid, (long)expires - 1);
+	ck_assert_int_eq(ret, 0);
+
+	ck_assert_int_eq(1, jwt_validate(jwt, jwt_valid));
+
+	// jwt is invalid when now >= expires
+	ret = jwt_valid_set_now(jwt_valid, (long)expires);
+	ck_assert_int_eq(ret, 0);
+
+	ck_assert_int_eq(0, jwt_validate(jwt, jwt_valid));
+
+	jwt_valid_free(jwt_valid);
+	__teardown_jwt();
+}
+END_TEST
+
+START_TEST(test_jwt_valid_headers)
+{
+	jwt_valid_t *jwt_valid = NULL;
+	int ret = 0;
+
+	// jwt is valid when iss in hdr matches iss in body
+	__setup_jwt();
+	jwt_add_header(jwt, "iss", "test");
+
+	ret = jwt_valid_new(&jwt_valid, JWT_ALG_NONE);
+	ck_assert_int_eq(ret, 0);
+	ck_assert(jwt_valid != NULL);
+
+	ret = jwt_valid_set_headers(jwt_valid, 1);
+	ck_assert_int_eq(ret, 0);
+
+	ck_assert_int_eq(1, jwt_validate(jwt, jwt_valid));
+
+	jwt_del_headers(jwt, "iss");
+
+	// jwt is invalid when iss in hdr does not match iss in body
+	jwt_add_header(jwt, "iss", "wrong");
+
+	ck_assert_int_eq(0, jwt_validate(jwt, jwt_valid));
+
+	jwt_del_headers(jwt, "iss");
+
+	// jwt is valid when checking hdr, but iss not replicated
+	ck_assert_int_eq(1, jwt_validate(jwt, jwt_valid));
+
+	jwt_valid_free(jwt_valid);
+	__teardown_jwt();
+}
+END_TEST
+
+
+//START_TEST(test_jwt_grants_json)
+//{
+//	const char *json = "{\"id\":\"FVvGYTr3FhiURCFebsBOpBqTbzHdX/DvImiA2yheXr8=\","
+//		"\"iss\":\"localhost\",\"other\":[\"foo\",\"bar\"],"
+//		"\"ref\":\"385d6518-fb73-45fc-b649-0527d8576130\","
+//		"\"scopes\":\"storage\",\"sub\":\"user0\"}";
+//	jwt_t *jwt = NULL;
+//	const char *val;
+//	char *json_val;
+//	int ret = 0;
+//
+//	ret = jwt_new(&jwt);
+//	ck_assert_int_eq(ret, 0);
+//	ck_assert(jwt != NULL);
+//
+//	ret = jwt_add_grants_json(jwt, json);
+//	ck_assert_int_eq(ret, 0);
+//
+//	val = jwt_get_grant(jwt, "ref");
+//	ck_assert(val != NULL);
+//	ck_assert_str_eq(val, "385d6518-fb73-45fc-b649-0527d8576130");
+//
+//	json_val = jwt_get_grants_json(NULL, "other");
+//	ck_assert(json_val == NULL);
+//	ck_assert_int_eq(errno, EINVAL);
+//
+//	json_val = jwt_get_grants_json(jwt, "other");
+//	ck_assert(json_val != NULL);
+//	ck_assert_str_eq(json_val, "[\"foo\",\"bar\"]");
+//
+//	jwt_free_str(json_val);
+//
+//	json_val = jwt_get_grants_json(jwt, NULL);
+//	ck_assert(json_val != NULL);
+//	ck_assert_str_eq(json_val, json);
+//
+//	jwt_free_str(json_val);
+//
+//	jwt_free(jwt);
+//}
+//END_TEST
```

Comment:
Either remove this, or enclose it in:

    #if 0
    ....code....
    #endif

As opposed to commenting it out.